lib/commands.js

```diff
@@ -2504,16 +2504,24 @@ commands.pressDeviceKey = commands.deviceKeyEvent;
 /**
  * pressKeycode(keycode, metastate, cb) -> cb(err)
  * metastate is optional.
+ * flags is optional for native android flag. Can set multiple falsg as Array. Flags have `FLAG_` prefix in http://developer.android.com/reference/android/view/KeyEvent.html
+ * e.g.: FLAG_CANCELED is `32` or `0x00000020`
  *
  * @jsonWire POST /session/:sessionId/appium/device/press_keycode
  */
 commands.pressKeycode = function() {
   var fargs = utils.varargs(arguments);
   var cb = fargs.callback,
       keycode = fargs.all[0],
-      metastate = fargs.all[1];
+      metastate = fargs.all[1],
+      flags = fargs.all[2];
   var data = {keycode: keycode};
   if(metastate) { data.metastate = metastate; }
+  if(flags) {
+    f = 0
```

Comment:
no reason to use a temporary variable instead of just `data.flags`, right? will save a line.